mods/timer-resolution-control.wh.cpp

```diff
@@ -215,25 +221,84 @@ BOOL Wh_ModInit(void)
     ULONG CurrentResolution;
     NTSTATUS status = ((NTSTATUS(WINAPI*)(PULONG, PULONG, PULONG))pNtQueryTimerResolution)(
         &MinimumResolution, &MaximumResolution, &CurrentResolution);
-    if (NT_SUCCESS(status)) {
-        Wh_Log(L"NtQueryTimerResolution: min=%f, max=%f, current=%f",
-            (double)MinimumResolution / 10000.0,
-            (double)MaximumResolution / 10000.0,
-            (double)CurrentResolution / 10000.0);
-        g_minimumResolution = MinimumResolution;
-        g_maximumResolution = MaximumResolution;
+    if (!NT_SUCCESS(status)) {
+        Wh_Log(L"NtQueryTimerResolution failed with status: 0x%X", status);
+        return FALSE;
     }
 
+    Wh_Log(L"NtQueryTimerResolution: min=%f, max=%f, current=%f",
+        (double)MinimumResolution / 10000.0,
+        (double)MaximumResolution / 10000.0,
+        (double)CurrentResolution / 10000.0);
+    g_minimumResolution = MinimumResolution;
+    g_maximumResolution = MaximumResolution;
+    g_lastDesiredResolution = CurrentResolution;    //TODO: During mod load there is a small race condition where the hook is not yet set, the g_lastDesiredResolution is already set, and then the program tries to set a new desired resolution which will not be saved into g_lastDesiredResolution.
+
     LoadSettings();
 
     Wh_SetFunctionHook((void*)pNtSetTimerResolution, (void*)NtSetTimerResolutionHook, (void**)&pOriginalNtSetTimerResolution);
 
     return TRUE;
 }
 
-void Wh_ModSettingsChanged(void)
+void EnforceLimits() 
+{
+    ULONG CurrentResolution;
+    NTSTATUS status = pOriginalNtSetTimerResolution(0, FALSE, &CurrentResolution);
+    if (status == STATUS_TIMER_RESOLUTION_NOT_SET) {
```

Comment:
If `STATUS_TIMER_RESOLUTION_NOT_SET` is returned, how about restoring the unset state on unload? According to the [ExSetTimerResolution documentation](https://learn.microsoft.com/en-us/windows-hardware/drivers/ddi/wdm/nf-wdm-exsettimerresolution) and [to ReactOS](https://github.com/reactos/reactos/blob/ea60890961dcdda7356a3c6cfed919fd20905f88/ntoskrnl/ex/time.c#L730), it can be done by setting a zero resolution, but I didn't test it.